src/Containers/MenuDetailsContainer/index.tsx

```diff
@@ -8,65 +8,84 @@ import Lunch from "../../Assets/Lunch.webp";
 import { toast } from "react-toastify";
-import { useNavigate } from "react-router-dom";
+import { useNavigate, useParams } from "react-router-dom";
+import SubscriptionPlanComponent from "../../Components/SubscriptionPlanComponent";
 
 function MenuDetailsContainer() {
+  type Params = {
+    id: string;
+    menuId: string;
+  };
+  const { id, menuId } = useParams<Params>();
+  const providerId = id || "";
+  const menuid = menuId || "";
   const navigate = useNavigate();
-  const [selectedCategory, setSelectedCategory] = useState<string>("BreakFast");
+  const [selectedCategory, setSelectedCategory] = useState<string>("Breakfast");
   const [menu, setMenu] = useState<MenuItem[]>([]);
   const [dailyModal, setDailyModal] = useState<boolean>(false);
+  const [subscriptionModal, setSubscriptionModal] = useState<boolean>(false);
   const [selectedCategories, setSelectedCategories] = useState<string[]>([]);
   const [selectedDate, setSelectedDate] = useState<string>("");
   const [totalAmount, setTotalAmount] = useState<number>(0);
 
   const categories = [
     {
       id: "0193ce2c-ab58-7b6f-8470-7462704e8638",
-      name: "BREAKFAST",
+      name: "Breakfast",
       image: BreakFast,
       description:
         "Morning Glory: A Feast to Jumpstart Your Day,A Feast to Jumpstart Your Day",
     },
     {
       id: "0193ce2d-5a2b-7a4c-b75a-aeaa23f3e6b2",
-      name: "LUNCH",
+      name: "Lunch",
       image: Lunch,
       description:
         "Midday Marvels: Lunchtime Delights Await,A Feast to Jumpstart Your Day",
     },
     {
       id: "0193ce2d-862c-7c4f-b3c2-7ee4c8e7113e",
-      name: "DINNER",
+      name: "Dinner",
       image: Dinner,
       description:
         "Dinner Delights: An Evening of Culinary Elegance ,A Feast to Jumpstart Your Day",
     },
   ];
 
-  const handlePay = () => {
+  const handlePay = (modalType: "daily" | "subscription") => {
     if (selectedCategories.length === 0) {
       toast.warn("Please select at least one category.");
       return;
     }
     navigate("order");
-    handleClose();
-  };
-
-  const handleClose = () => {
-    setDailyModal(false);
+    handleClose(modalType);
   };
 
-  const handleOpen = () => {
-    setDailyModal(true);
+  const handleClose = (modalType: "daily" | "subscription") => {
+    if (modalType === "daily") {
+      setDailyModal(false);
+    } else if (modalType === "subscription") {
+      setSubscriptionModal(false);
+    }
+    setSelectedCategories([]);
+    setSelectedDate("");
+    setTotalAmount(0);
   };
 
   useEffect(() => {
     if (selectedDate && selectedCategories.length > 0) {
-      CalculateTotal(selectedDate, selectedCategories).then((response) => {
+      const is_subscription = dailyModal ? false : true;
+      CalculateTotal(
+        selectedDate,
+        selectedCategories,
+        providerId,
+        menuid,
+        is_subscription
+      ).then((response) => {
         setTotalAmount(response.data.result);
```

Comment:
@fathimathusuhaila if response data is empty it will through error , so give a condition like if(response && response.data && response.data.result) {}